devito/types/dense.py

```diff
@@ -308,16 +308,29 @@ def shape_global(self):
     @cached_property
     def _size_outhalo(self):
         """Number of points in the outer halo region."""
+
         if self._distributor is None:
+            # Computational domain is not distributed and hence the outhalo
+            # and inhalo correspond
             return self._size_inhalo
 
-        left = [self._distributor.glb_to_loc(d, i, LEFT, strict=False)
-                for d, i in zip(self.dimensions, self._size_inhalo.left)]
-        right = [self._distributor.glb_to_loc(d, i, RIGHT, strict=False)
-                 for d, i in zip(self.dimensions, self._size_inhalo.right)]
+        left = [abs(min(i.loc_abs_min-i.glb_min-j, 0)) if i and not i.loc_empty else 0
+                for i, j in zip(self._decomposition, self._size_inhalo.left)]
+        right = [max(i.loc_abs_max+j-i.glb_max, 0) if i and not i.loc_empty else 0
+                 for i, j in zip(self._decomposition, self._size_inhalo.right)]
 
         sizes = tuple(Size(i, j) for i, j in zip(left, right))
 
+        if self._distributor.is_parallel and any(left) > 0 or any(right) > 0:
+            try:
+                if not self._distributor.boundary_rank:
+                    warning("A space order of %d has been set but the current rank "
```

Comment:
I think we should include the "halo size" in this msg. One could indeed override the default halo that a given space order gives you through "Function(...., halo=...)".

I also wonder whether (nitpicking) "domain size" sounds better than "spatial extent" ? we use "extent" for physical quantities and "size" for discrete ones

Reply:
Oki, will edit.